src/main/java/com/novemberain/quartz/mongodb/db/MongoConnectorBuilder.java

```diff
@@ -132,15 +151,55 @@ private MongoClientOptions createOptions() {
         if (threadsAllowedToBlockForConnectionMultiplier != null) {
             optionsBuilder.threadsAllowedToBlockForConnectionMultiplier(threadsAllowedToBlockForConnectionMultiplier);
         }
-        if (enableSSL != null) {
-            optionsBuilder.sslEnabled(enableSSL);
-            if (sslInvalidHostNameAllowed != null) {
-                optionsBuilder.sslInvalidHostNameAllowed(sslInvalidHostNameAllowed);
+        SSLContext sslContext = getSSLContext();
+        if (sslContext == null) {
+            if (enableSSL != null) {
+                optionsBuilder.sslEnabled(enableSSL);
+                if (sslInvalidHostNameAllowed != null) {
+                    optionsBuilder.sslInvalidHostNameAllowed(sslInvalidHostNameAllowed);
+                }
             }
+        } else {
+            optionsBuilder.sslEnabled(true);
+            optionsBuilder.sslContext(sslContext);
         }
         return optionsBuilder.build();
     }
 
+    private SSLContext getSSLContext() throws SchedulerConfigException {
+        try {
+            KeyStore trustStore = loadKeyStore(trustStorePath, trustStorePassword, trustStoreType);
+            KeyStore keyStore = loadKeyStore(keyStorePath, keyStorePassword, keyStoreType);
+            if (trustStore == null && keyStore == null) {
+                return null;
+            }
+            SSLContextBuilder sslContextBuilder = SSLContexts.custom();
+            if (trustStore != null) {
+                sslContextBuilder.loadTrustMaterial(trustStore, null);
+            }
+            if (keyStore != null) {
+                sslContextBuilder.loadKeyMaterial(keyStore,
+                        StringUtils.isBlank(keyStorePassword) ? null : keyStorePassword.toCharArray());
```

Comment:
Why doesn't the `loadTrustMaterial` above have the same password logic?

Reply:
Because its second parameter isn't the password, it's a `TrustStrategy`.
So, basically, I've copied the logic from https://github.com/nuxeo/nuxeo/blob/master/nuxeo-runtime/nuxeo-runtime-mongodb/src/main/java/org/nuxeo/runtime/mongodb/MongoDBConnectionHelper.java#L144.

Reply:
Ok sorry